lib/compiler/src/beam_types.erl

```diff
@@ -973,8 +989,27 @@ lub(#t_tuple{size=SzA,elements=EsA}, #t_tuple{size=SzB,elements=EsB}) ->
     Sz = min(SzA, SzB),
     Es = lub_tuple_elements(Sz, EsA, EsB),
     #t_tuple{size=Sz,elements=Es};
-lub(_T1, _T2) ->
-    %%io:format("~p ~p\n", [_T1,_T2]),
+lub(T1, T2) ->
+    %%io:format("~p ~p\n", [T1,T2]),
+    case is_other(T1) andalso is_other(T2) of
+        true -> other;
+        false -> any
+    end.
+
+is_other(other) -> true;
```

Comment:
I think we need to add this to `beam_types_SUITE`, ensuring that `is_other(T) =:= true` can never meet any other types and vice versa.

An alternative is to express this in terms of `meet/2`, meeting the most general union type where `other=none`. If the result of that is `none`, we know that it is `other`.

Reply:
I went for your alternative approach. Does that mean that the existing test case will find bug in the handling of `other` or should I still try to write some new test case?

Reply:
Yeah, I think the existing model would find related errors.